src/decoder.ts

```diff
@@ -64,3 +72,53 @@ export function decode<T = unknown>(
   }
   return ret as T;
 }
+
+/**
+ * Decode CBOR Sequence bytes to major-type/additional-information/value tuples.
+ */
+export class Sequence {
+  #seq: ValueGenerator;
+  #peeked: MtAiValue | undefined;
+
+  public constructor(src: Uint8Array | string, options: DecodeOptions = {}) {
+    const stream = new DecodeStream(src, normalizeOptions(options));
+    this.#seq = stream.seq();
+  }
+
+  /** Peek at the next tuple, allowing for later reads. */
+  public peek(): MtAiValue | undefined {
+    if (!this.#peeked) {
+      this.#peeked = this.#next();
+    }
+    return this.#peeked;
+  }
+
+  /** Read the next tuple. */
+  public read(): MtAiValue | undefined {
+    const mav = this.#peeked ?? this.#next();
+    this.#peeked = undefined;
+    return mav;
+  }
+
+  /** Iterate over all tuples. */
+  public *[Symbol.iterator](): Generator<MtAiValue, void, undefined> {
+    while (true) {
+      const tuple = this.read();
+
+      if (!tuple) {
+        return;
+      }
+
+      yield tuple;
+    }
+  }
+
+  #next(): MtAiValue | undefined {
+    const {value: tuple, done} = this.#seq.next();
+    if (done) {
+      return undefined;
+    }
+
+    return tuple;
```

Comment:
I wonder if we should call `Object.freeze` on these, so that you can't peek, modify the result, and cause the next read to be different than expected.

Reply:
If you're peeking and then modifying the value you're really shooting yourself in the foot.

Reply:
I guess I'm wondering how you're going to use this.  Is it possible this is a security issue?

Reply:
In Concordance, I'm encoding aspects of an object as a sequence. There's a marker item for each type of aspect (named property, symbol property, iterator value, etc). So when decoding named properties, if I see the symbol property marker I know to stop decoding named properties. This is easiest if I can peek at the next item.

Reply:
Close enough.  We'll keep an eye on how this is used, and can always add a freeze if desired later.